### ISSUE TITLE
On iOS Launcher.OpenAsync presents Open-In/ShareSheet Menu but closes prematurely or is unresponsive after clicking on one of the options

### ISSUE THREAD

**Issue by fadihrezq** [xamarin/Essentials#1376]
When trying to open a file I downloaded into my app's Documents folder, using the Launcher.OpenAsync, it presents the "Open In Menu" (ShareSheet menu) but clicking on any of the options does not do anything or sometimes just closes the menu.

Launcher.OpenAsync works normally in Android but this issue appears in iOS.

After several days of trying different things I finally created my own DependencyInjection iOS implementation of the Launcher.OpenAsync code (copied from the repo here) and after tinkering around with it, I found that if I keep a reference of the UIDocumentInteractionController and a reference of the Launcher implementation, then it works.

My guess is that the UIDocumentInteractionController still needs to be alive when the user clicks the icon of whatever app needs to handle the file. In the Xamarin.Essentials Launcher code, the UIDocumentInteractionController is instantiated as a local variable in the OpenAsync method and after it presents the "Open In Menu",  the function returns and the interaction controller goes out of scope. So when the user tries to interact with the menu, the interaction controller reference is no longer valid and this is what causes the issue.

Here is a sample of my code:

In the iOS implementation:
```
public class IOSShareSheetManager : IShareSheetManager
{
     UIDocumentInteractionController controller = null;
     public Task Open(OpenFileRequest request)
     {
           var fileUrl = NSUrl.FromFilename(request.File.FullPath);

           controller = UIDocumentInteractionController.FromUrl(fileUrl);
           controller.Uti = request.File.ContentType;

          ....

          controller.PresentOpenInMenu(rect.Value, vc.View, true);

          return Task.CompletedTask;
     }
}
```

shared code usage:
```
if (shareSheetManager == null)
     shareSheetManager = DependencyService.Get<IShareSheetManager>();
await shareSheetManager.Open(fileRequest);
```

I also believe this also may be related to the following issues: 

 #https://github.com/xamarin/Essentials/issues/1166
 #https://github.com/xamarin/Essentials/issues/1348

I am using:
Xamarin.Forms v4.7
Xamarin.Essentials 1.5.3.2

**Comment by Redth** [comment]
Should be fixed in #1380

**Comment by Aguirrehub25** [comment]
you can implement the classes in visual studio 2019 ? 
I have the same problem displaying images on ios
